js/src/setup-mc/setup-stepper/setup-paid-ads/setup-paid-ads.js

```diff
@@ -24,37 +27,68 @@ import PaidAdsFeaturesSection from './paid-ads-features-section';
 import GoogleAdsAccountSection from './google-ads-account-section';
 import AudienceSection from '.~/components/paid-ads/audience-section';
 import BudgetSection from '.~/components/paid-ads/budget-section';
+import BillingCard from '.~/components/paid-ads/billing-card';
 import validateForm from '.~/utils/paid-ads/validateForm';
 import { getProductFeedUrl } from '.~/utils/urls';
-import { GUIDE_NAMES, GOOGLE_ADS_ACCOUNT_STATUS } from '.~/constants';
 import { API_NAMESPACE } from '.~/data/constants';
+import {
+	GUIDE_NAMES,
+	GOOGLE_ADS_ACCOUNT_STATUS,
+	GOOGLE_ADS_BILLING_STATUS,
+} from '.~/constants';
 
-const { CONNECTED } = GOOGLE_ADS_ACCOUNT_STATUS;
-
-function PaidAdsSectionsGroup( { onCampaignChange } ) {
+function PaidAdsSectionsGroup( { onStatesReceived } ) {
 	const { googleAdsAccount } = useGoogleAdsAccount();
 	const { data: targetAudience } = useTargetAudienceFinalCountryCodes();
+	const { billingStatus } = useGoogleAdsAccountBillingStatus();
 
-	if ( ! targetAudience ) {
-		return <GoogleAdsAccountSection />;
-	}
+	const onStatesReceivedRef = useRef();
+	onStatesReceivedRef.current = onStatesReceived;
 
-	const handleChange = ( _, values, isValid ) => {
-		onCampaignChange( { ...values, isValid } );
+	const initialValues = {
+		amount: 0,
+		countryCodes: targetAudience,
 	};
+	const [ campaign, setCampaign ] = useState( {
+		...initialValues,
+		isValid: ! Object.keys( validateForm( initialValues ) ).length,
+		isReady: false,
+	} );
+
+	const isBillingCompleted =
+		billingStatus?.status === GOOGLE_ADS_BILLING_STATUS.APPROVED;
+
+	// Watch the `isBillingCompleted` in order to ensure the parent component can
+	// continue the setup from a middle state. For example, refresh the current page
+	// after the billing setup is finished.
+	useEffect( () => {
+		onStatesReceivedRef.current( {
```

Comment:
❓  why we need to use REF here and not just `onStatesReceived` prop?

Reply:
❓  Also could you elaborate more the comment added here? 

I understand that we are trying to prevent some kind of issue in regards Step 4??? But not sure if I understand it fully.

Reply:
> why we need to use REF here and not just `onStatesReceived` prop?

The `onStatesReceived` callback is not a wanted deps for triggering this useEffect but it needs to be invoked with the latest reference within this useEffect.

Reply:
> Also could you elaborate more the comment added here?
> 
> I understand that we are trying to prevent some kind of issue in regards Step 4??? But not sure if I understand it fully.

I rephrased the code comment in https://github.com/woocommerce/google-listings-and-ads/pull/1677/commits/bda14476da754ecdd2af833603a9ccc7ea4766e9. Let me know if there is anything still confusing or unclear.